pysph/cpy/translator.py

```diff
@@ -532,6 +539,26 @@ def visit_FunctionDef(self, node):
         args = self._get_function_args(node)
         body = '\n'.join(self._indent_block(self.visit(item))
                          for item in self._remove_docstring(node.body))
+        local_decl = ''
+        if self._local_decl:
+            decls = ['extern LOCAL_MEM float shared_buff[];']
+            for arg, dtype in self._local_decl.items():
+                if len(decls) == 1:
+                    local_decl = ('%(dtype)s* %(arg)s = '
+                                  '(%(dtype)s*) shared_buff;')
+                    local_decl = local_decl % {'dtype': dtype, 'arg': arg}
+                    decls.append(local_decl)
+                    prev_arg = arg
+                else:
+                    local_decl = ('%(dtype)s* %(arg)s = (%(dtype)s*) '
+                                  '&%(prev_arg)s[size_%(prev_arg)s];')
+                    local_decl = local_decl % {'dtype': dtype, 'arg': arg,
+                                               'prev_arg': prev_arg}
+                    decls.append(local_decl)
```

Comment:
This whole block of code is not at all intentional and it is very difficult to see what you are trying to do.  I think it really needs to be refactored.